pokemon-checker/src/services/PokemonService.ts

```diff
@@ -93,7 +93,8 @@ export class PokemonService {
     } else {
       const data = await response.json();
       const pokemonPayload = data as IPokemonData;
-      return this.factory.createPokemon(pokemonPayload);
+      const pokemon = this.factory.createPokemon(pokemonPayload);
+      return pokemon;
```

Comment:
These 2 lines can probably be reduced back to this:

```suggestion
      return this.factory.createPokemon(pokemonPayload);
```